[docs] Adds workflows syntax doc (#34649)
expo/expo: docs/pages/eas/workflows/examples.mdx

---

import { BookOpen02Icon } from '@expo/styleguide-icons/outline/BookOpen02Icon';

import { BoxLink } from '~/ui/components/BoxLink';
import { Collapsible } from '~/ui/components/Collapsible';
import { Terminal } from '~/ui/components/Snippet';

The following workflows are examples of how you can use EAS Workflows to automate your development and release processes. They can help you and your team develop, review each other's PRs, and release changes to your users.

## Development builds workflow

[Development builds](/develop/development-builds/introduction/) are specialized builds of your project that include Expo's developer tools. These types of builds include all native dependencies inside your project, enabling you to run a production-like build of your project on a simulator, emulator, or a physical device.

<Collapsible summary="Prerequisites">

To get started, you'll need to configure your project and devices to build and run development builds.

<BoxLink
  title="Development build prerequisites"
  description="Get your project ready for development builds."
  href="/develop/development-builds/create-a-build/#prerequisites"
  Icon={BookOpen02Icon}

> />

Reviewer: This could lead to some confusion and a lengthy path, I think. Because the prerequisites leed to EAS Build page. Instead can we link Android and iOS  environment sections like this one: https://docs.expo.dev/get-started/set-up-your-environment/?mode=development-build#set-up-an-android-device-with-a-development-build?

Reviewer: Reading the content below, it is mostly installing EAS CLI and dev client library since the build profile stuff is already covered, maybe we can just include those steps here.

Author: Thanks for noticing this. I am going to include four links (ios device, android device, ios simulator, android emulator), since all of them require different steps (like having a developer account, having xcode, etc)

After you've configured your project and devices, add the following build profiles to your **eas.json** file.

```json eas.json
{
  "build": {
    "development": {
      "developmentClient": true,
      "distribution": "internal"
    },
    "development-simulator": {
      "developmentClient": true,
      "distribution": "internal",
      "ios": {
        "simulator": true
      }
    }
  }
}
```

</Collapsible>

The following workflow creates a build for each platform and for both physical devices, Android emulators, and iOS simulators. They all will run in parallel.

```yaml .eas/workflows/development-build.yml
name: Development Build

jobs:
  android_development_build:
    type: build
    params:
      platform: android
      profile: development
  ios_device_development_build:
    type: build
    params:
      platform: ios
      profile: development
  ios_simulator_development_build:
    type: build
    params:
      platform: ios
      profile: development-simulator
```

> Run the workflow above with:

Reviewer: ```suggestion
Run the above workflow with:
```

<Terminal cmd={['$ eas workflow:run .eas/workflows/development-build.yml']} />

## PR preview workflow

Once you've made changes to your project, you can share a preview of your changes with your team by publishing a [preview update](/review/share-previews-with-your-team/).

You can access preview updates in the development build UI and through scannable QR codes on the Expo dashboard. When publishing a preview on every commit, your team can review changes without pulling the latest changes and running them locally.

<Collapsible summary="Prerequisites">

> Your project needs have [EAS Update](/eas-update/introduction/) setup to publish preview updates. You can set up your project with:

Reviewer: ```suggestion
Your project needs to have [EAS Update](/eas-update/introduction/) setup to publish preview updates. You can set up your project with:
```

<Terminal cmd={['$ eas update:configure']} />

> After that, you'll need to create new [development builds](/develop/development-builds/introduction/).

Reviewer: ```suggestion
After that, you'll need to create new [development builds](/develop/development-builds/create-a-build/).
```

I think linking how to create a dev build page might be better in this context.

</Collapsible>

The following workflow publishes a preview update for every commit on every branch.

```yaml .eas/workflows/pr-preview.yml
name: Publish Preview Update

on:
  push:
    branches: ['*']

jobs:
  publish_preview_update:
    type: update
    params:
      branch: ${{ github.ref || 'test' }}
```

## Deploy to production workflow

> When you're ready to deliver changes to your users, you can build and submit to the app stores or you can send and over-the-air update. The following workflow detects if you need new builds, and if so, it sends them to the app stores. If new builds are not required, it will send an over-the-air update.

Reviewer: ```suggestion
When you're ready to deliver changes to your users, you can build and submit to the app stores or you can send an over-the-air update. The following workflow detects if you need new builds, and if so, it sends them to the app stores. If new builds are not required, it will send an over-the-air update.
```
